docs/guides/create-composable-components.stories.mdx

```diff
@@ -0,0 +1,187 @@
+import { Meta } from "@storybook/addon-docs";
+
+<Meta title="Guides/Creat Composable Components" />
+
+# Create Composable Components
+
+### Why Composability Matters
+
+**Feedback from Atlantis consumers has highlighted that**:
+
+- Developers want to make minor modifications without requesting feature
+  additions.
+- Designers need more flexibility to tailor components to specific use cases.
+- Current rigidity leads to workarounds, custom builds, and decreased adoption
+  of Atlantis.
+
+**Key Goals**:
+
+- Enhance flexibility while maintaining consistency.
+- Empower teams to build their own variations using reusable building blocks.
+
+### Types of Components and Their Customization Options
+
+1. **Primitive Components**:  
+   Examples: [Icon](..?path=/docs/components-images-and-icons-icon--docs),
+   [Avatar](..?path=/docs/components-images-and-icons-avatar--docs)
+
+   These components do not accept children and have minimal customization.
+
+2. **Simple Components**:  
+   Examples: [Button](..?path=/docs/components-actions-button--docs),
+   [Link](..?path=/docs/components-text-and-typography-link--docs)
+
+   These allow basic customization through props or limited children, like
+   `ReactNode` or specific child types.
+
+3. **Compound Components**:  
+   Examples: [Chip](..?path=/docs/components-selections-chip--docs),
+   [DataList](..?path=/docs/components-lists-and-tables-datalist--docs)
+
+   These are part of larger compositions and may validate their children types
+   to maintain internal consistency.
+
+4. **Complex Components**:  
+   Examples:
+   [Autocomplete](..?path=/docs/components-forms-and-inputs-autocomplete--docs),
+   [Combobox](..?path=/docs/components-selections-combobox--docs)
+
+   These often have strict APIs, with customization limited to specific props.
+   Internal UI and behaviors are not easily modifiable.
+
+## Path Forward: Favoring a Custom Render Function
+
+To address these challenges, we are adopting a **custom render function**
+approach as a preferred pattern for composability.
+
+#### Named Render Prop: `customRender____`
+
+This works by providing a named render prop (e.g., `customRenderItem`) that
+allows consumers to inject their own UI while preserving default behaviors. For
+example:
+
+```tsx
+const renderProductItem = item => (
+  <Flex template={["shrink", "grow"]} align="start">
+    <Text>{item.price}</Text>
+    <Heading level={4}>{item.name}</Heading>
+  </Flex>
+);
+
+export const CustomRenderExample = () => (
+  <List items={items} customRenderItem={renderProductItem} />
+);
+```
+
+**Why This Approach?**
+
+- **Clarity**: The `customRender` prefix clearly communicates the desire to do
+  advanced customization.
+- **Flexibility**: Supports both simple and complex components without major
+  refactoring.
+- **Trackability**: Easy to search for and monitor across codebases.
```

Comment:
while true, and a selling point for us - I'm not sure this is something we need to tell others.